database/gormdb_submission_test.go

```diff
@@ -41,6 +41,64 @@ func setupCourseAssignment(t *testing.T, db database.Database) (*qf.User, *qf.Co
 	return user, course, assignment
 }
 
+func TestGormDBCreateSubmissionWithAutoApprove(t *testing.T) {
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	user, _, assignment := setupCourseAssignment(t, db)
+
+	assignment.AutoApprove = true
+	assignment.ScoreLimit = 1
+
+	if err := db.UpdateAssignments([]*qf.Assignment{assignment}); err != nil {
+		t.Fatal(err)
+	}
+
+	// create a new submission, ensure that it is auto-approved
+
+	submission := &qf.Submission{
+		AssignmentID: assignment.ID,
+		UserID:       user.ID,
+		Score:        1,
+	}
+	if err := db.CreateSubmission(submission); err != nil {
+		t.Error(err)
+	}
+
+	want := &qf.Submission{
+		ID:           submission.ID,
+		AssignmentID: assignment.ID,
+		UserID:       user.ID,
+		Score:        1,
+		Grades:       []*qf.Grade{{UserID: user.ID, SubmissionID: submission.ID, Status: qf.Submission_APPROVED}},
+	}
+
+	if diff := cmp.Diff(submission, want, protocmp.Transform()); diff != "" {
```

Comment:
I prefer to use `got` to make it clear that it is the result of the operation we are testing.